File: 2461-maximum-sum-of-distinct-subarrays-with-length-k/2461-maximum-sum-of-distinct-subarrays-with-length-k.py
from typing import List

class Solution:
    def maximumSubarraySum(self, nums: List[int], k: int) -> int:
        seen = set()
        window_sum = 0
        left = 0
        max_sum = 0

        for right in range(len(nums)):
            # If duplicate found, shrink window from left
            while nums[right] in seen:
                seen.remove(nums[left])
                window_sum -= nums[left]
                left += 1

            # Add current number
            seen.add(nums[right])
            window_sum += nums[right]

            # When window size exceeds k, shrink from left
            if right - left + 1 > k:
                seen.remove(nums[left])
                window_sum -= nums[left]
                left += 1

            # If window size == k → valid candidate
            if right - left + 1 == k:
                max_sum = max(max_sum, window_sum)

        return max_sum

File: 2461-maximum-sum-of-distinct-subarrays-with-length-k/2461-maximum-sum-of-distinct-subarrays-with-length-k.java
class Solution {
    public long maximumSubarraySum(int[] nums, int k) {
        int l=0;
        long ans=0,sum=0;
        HashSet <Integer> mp = new HashSet<>();
        for(int r = 0;r<nums.length;r++){
            while(mp.contains(nums[r])){
                sum-=nums[l];
                mp.remove(nums[l]);
                l++;
            }
            sum+=nums[r];
            mp.add(nums[r]);
            while(r-l+1>k && l<=r){
                sum-=nums[l];
                mp.remove(nums[l]);
                l++;
            }
            if(r-l+1==k) ans = Math.max(ans,sum);
        }
    return ans;
    }
}